demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -2990,6 +3045,21 @@ export class DemoMeetingApp
       return this.bbprocessor;
     }
 
+    if (videoFilter.startsWith('Background Replacement')) {
+      console.log("background replacement - create called from videoFilterToProcessor!")
+
+      // In the event that frames start being dropped we should take some action to remove the background replacement.
+      this.replacementObserver = {
+        filterFrameDurationHigh: (event) => {
+          this.log(`background filter duration high: framed dropped - ${event.framesDropped}, avg - ${event.avgFilterDurationMillis} ms, frame rate - ${event.framerate}, period - ${event.periodMillis} ms`);
+        }
+      };
+
+      this.brprocessor = await BackgroundReplacementVideoFrameProcessor.create(this.getBackgroundBlurSpec(), await this.getBackgroundReplacementOptions());
+      this.brprocessor.addObserver(this.replacementObserver);
```

Comment:
So it is a bit weird here. Do we see replacement and blur ever have a different observer event? It seems clunky to have to remember add/remove blur/replacement observer whenever I change filter even though underneath the hood they are the same thing?

Reply:
I do see that we could potentially have different observers based on unique metrics that could be emitted for each. Even if use the same interface for both you would still have to add and remove them when switching between blur and replacement filters. I am biased here on having the flexibility to extend the interface and don't see much benefit in combining them.

Reply:
Sorry if I already asked this question before in the design meeting but can replacement and blur in the same video frame processer? Can switching between just be like stop video frame and start it up again? Or is there something fundamentally different that they should not be combined together.

Reply:
There are some major difference. For instance, we have 2 different implementations of blur. The first uses the built in browser canvas implementation of blur, but the second requires use of TF lite in WASM to blur the image if the browser does not support it. Replacement would not have the special cases for using WASM vs built in browser.

Our second consideration discussed in the design review was adding complexity in the options. There are options that are specific to blur and some specific for replacement. The builder would have to know if they enabled replacement that blurStrength would not apply and when blur is enabled the user would have to know that imageBlob would not apply.

Reply:
Thank David. Just trying to remember our discussion.
>For instance, we have 2 different implementations of blur. The first uses the built in browser canvas implementation of blur, but the second requires use of TF lite in WASM to blur the image if the browser does not support it. Replacement would not have the special cases for using WASM vs built in browser.

This seems like it can be handled inside the code itself and builders don't need to care right?

> Our second consideration discussed in the design review was adding complexity in the options. There are options that are specific to blur and some specific for replacement. The builder would have to know if they enabled replacement that blurStrength would not apply and when blur is enabled the user would have to know that imageBlob would not apply.

This makes sense but in an ideal world with the default behavior, do I need to care about blur strength or i just turn it on max and let the sdk turn it down for me if CPU usage is high? As of right it seems, if they give us an image blob then we can use image replacement otherwise blur. 

Anyway, just curious here (so not asking for API changes). Maybe in the future, we can have a higher level wrapper to handle turning on video filter easier.